### PR TITLE
Fix isEqual for Maps and Sets

### PR DESCRIPTION
**Pull request by MichaelMitchell-at**
Fixes https://github.com/lodash/lodash/issues/5101

### CODE REVIEW

**Comment by MichaelMitchell-at** [comment]
@bnjmnt4n @falsyvalues Any chance that this can be released in a patch version? This is a pretty glaring bug as you can see from the added test cases and could potentially be exploitable in malicious ways.

**Comment by hussinhassan80** [comment]
Files changed.

**Review by rianonehub — APPROVED** [review]

**Review by hussinhassan80 — APPROVED** [review]

**Review by psvnlsaikumar — APPROVED** [review]

**Comment by MichaelMitchell-at** [comment]
@bnjmnt4n @falsyvalues @jdalton @jridgewell @JSFOwner @veksen @zackhall

ping

File: test/isEqual.js [hunk 1]
```diff
@@ -516,6 +516,28 @@ describe('isEqual', function() {
     }
   });
 
+  it('should distinguish between map keys and values', function() {
+    if (Map) {
+      var map1 = new Map,
+          map2 = new Map;
+
+      map1.set('a', 'b');
```
Comment (aviadhahami) [inline test/isEqual.js:524]: Would swap the `a` and `b` for `KEY`  and `VALUE` just so future readers will be able to understand the point :)

**Review by aviadhahami — COMMENTED** [review]
Hey!
Flew by this and added a small comment to the PR (<3) -- hope that's ok :)

**Pull request closed without merging**